src/Symfony/Component/Yaml/Inline.php

```diff
@@ -216,30 +227,39 @@ public static function isHash(array|\ArrayObject|\stdClass $value): bool
      */
     private static function dumpArray(array $value, int $flags): string
     {
+        $numericKeysAsString = Yaml::DUMP_NUMERIC_KEY_AS_STRING & $flags;
```

Comment:
Why do you separate it here in a var you only use once?

And this part you have three times now, perhaps it's an idea to separate it into a function. 

Idea:
```
$keyFlags = $valueFlags = $flags;
```
always and then only manipulating the valueFlags:
```
if (Yaml::DUMP_NUMERIC_KEY_AS_STRING & $valueFlags) {
  $valueFlags -= Yaml::DUMP_NUMERIC_KEY_AS_STRING;
}

Reply:
Thanks for your suggestion ! However with other remarks I rework PR, is now simplier